src/compiler/checker.ts

```diff
@@ -11225,7 +11225,18 @@ module ts {
 
         function checkGrammarFunctionLikeDeclaration(node: FunctionLikeDeclaration): boolean {
             // Prevent cascading error by short-circuit
-            return checkGrammarModifiers(node) || checkGrammarTypeParameterList(node, node.typeParameters) || checkGrammarParameterList(node.parameters);
+            return checkGrammarModifiers(node) || checkGrammarTypeParameterList(node, node.typeParameters) || checkGrammarParameterList(node.parameters) || checkGrammarArrowFunction(node);
+        }
+
+        function checkGrammarArrowFunction(node: FunctionLikeDeclaration): boolean {
+            if (node.kind === SyntaxKind.ArrowFunction && (<ArrowFunction>node).arrow) {
+                var arrowFunction = <ArrowFunction>node;
+                var sourceFile = getSourceFileOfNode(node);
+                if (getLineAndCharacterOfPosition(sourceFile, getTokenPosOfNode(arrowFunction.arrow, sourceFile)).line !== getLineAndCharacterOfPosition(sourceFile, arrowFunction.parameters.end).line) {

```

Comment:
I'd rather you just extracted the lines into individual variables and compared them